### ISSUE TITLE
Firefox 57 - need a new extension

### ISSUE THREAD

**Issue by bobharrisbtt** [docwhat/itsalltext#97]
Firefox 57 breaks the current It's All Text
A new Firefox 57 compatible extension is needed, desired, wanted, Please :-)

**Comment by xOneca** [comment]
Please, see [README](/docwhat/itsalltext/blob/6189036/README.md):
> With the Firefox 57, It's All Text! has stopped working and fixing it would require throwing everything out and starting again from scratch.

We've lost IAT forever... :cry:

**Comment by Li-An** [comment]
You can test Waterfox, an alternative for Firefox.

**Comment by gibus** [comment]
After having bad experiences with some other alternative (GhostText, Textern), I'm using withExEditor for a couple of days and eventually found it gives same functionalities as IAT.

You have to install a nodejs host (see https://github.com/asamuzaK/withExEditorHost. The only difficulty is to have nodejs > v8.9.0 (for Debian Stretch or Ubuntu Xenial, you've got to installing it from https://nodejs.org/en/download/package-manager/).

You've got an install and configure scripts to run and then the host is transparently launched by Firefox, and a right click on any textarea gives allows you to "edit with XXX".

**Comment by Li-An** [comment]
Thanks for the alternative - but it’s harsh to install.

**Comment by bobharrisbtt** [comment]
Yes, thank you gibus for the withExEditor pointer.  And I agree with Li-An, it is tedious and annoying to setup on my Mac.  I ended up using MacPorts.org to install nodejs8 and npm5 just so I could install the withExEditorHost, to make withExPorts happy using MacVim.

I did have to set the MacVim preferences to quit after the last window closes, otherwise I ended up with multiple instances of MacVim running as I edited more and more browser text boxes.

**Comment by sputnick-dev** [comment]
The better ever extension is down :/ Sad !

AFAIK, we must bind a service on a port to work with withExEditor ?!

**Comment by docwhat** [comment]
Running a stand-alone service is the only way to work with the new security model.

**Comment by lrineau** [comment]
> Running a stand-alone service is the only way to work with the new security model.

That is not true. The communication protocol can also be integrated directly in the editor. [Emacs] has a package, [Atom] as well.

[Atom]: https://github.com/tuvistavie/atomic-chrome/
[Emacs]: https://github.com/alpha22jp/atomic-chrome

**Comment by arpepper** [comment]
**Re:** gibus commented on Nov 25, 2017...

It might be nice to mention this (withExEditor) in README.md; I wasted a lot of time getting effectively nowhere with GhostText.


withExEditorHost basically just worked for me (Ubuntu 18.04 actually in an lxc container)

**My notes...**


On an 18.04 container "apt-get install npm" gets the right node.js

Needs
https://github.com/asamuzaK/withExEditorHost

    Host setup from source code
    > cd path/to/withExEditorHost
    > npm run setup


(Caveat:  withExEditor won't seem to work on our mattermost site).

**Comment by gibus** [comment]
New versions of withExEditor need nodejs 10 which is not in ubuntu/debian repo. You've got to install it:
curl -sL https://deb.nodesource.com/setup_10.x | sudo -E bash -
sudo apt-get install -y nodejs
